scripts/linters/custom_eslint_checks/rules/no-then-method.js

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Rule to disallow using .then() method on a variable.
+ */
+
+'use strict';
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Rule to disallow using .then() method on a variable',
+      category: 'Best Practices',
+      recommended: true
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      avoidThenMethod: 'Please avoid using .then() method'
```

Comment:
Please add instruction on what developers need to do instead?

Reply:
Done!   As far as I understand, `.then()` is used with `Promise` objects. So, an alternative to using `.then()` would be to use `async/await`. I've changed the error message to `Please do not use .then(), consider async/await instead`. What do you think?